### PR TITLE
Added validation checking that cliName and ID match

### PR DESCRIPTION
**Pull request by israelpoli**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/46706

## Description
Added a new validation to **validate** command to verify that *cliName* and *id* keys
of the incident field or the indicator field are matches.

## Must have
- [x] Tests

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/field_base_validator.py [hunk 1]
```diff
@@ -214,7 +214,28 @@ def is_valid_cli_name(self) -> bool:
...
+
+    def is_cli_name_and_id_are_matches(self):
+        """
+        Checks if the field cliName matched to id.
+        Returns:
+            (bool): True if cliName matched to id, False otherwise.
+        """
+        cli_name = self.current_file.get('cliName', '')
+        _id = self.current_file.get('id', '')
+        if _id.startswith('incident'):
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/field_base_validator.py:228]: does `incident_` suit better?

File: demisto_sdk/commands/common/hook_validations/field_base_validator.py [hunk 2]
```diff
@@ -214,7 +214,28 @@ def is_valid_cli_name(self) -> bool:
...
+            _id = re.split('incident', _id, maxsplit=1)[1].lower().replace('_', '')
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/field_base_validator.py:229]:
```suggestion
            _id = _id[len('incident')+1:]
```

File: demisto_sdk/commands/common/hook_validations/field_base_validator.py [hunk 3]
```diff
@@ -214,7 +214,28 @@ def is_valid_cli_name(self) -> bool:
...
+        elif _id.startswith('indicator'):
+            _id = re.split('indicator', _id, maxsplit=1)[1].lower().replace('_', '')
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/field_base_validator.py:231]:
```suggestion
            _id = _id[len('indicator')+1:]
```

File: demisto_sdk/commands/common/tests/field_validator_test.py [hunk 4]
```diff
@@ -190,6 +190,50 @@ def test_is_cli_name_is_builtin_key_invalid(self, cliname, group):
...
+        ("incident_testfortest", "testfortest"),
+        ("incident_test_for_test", "testfortest"),
+        ("indicator_test_for_test", "testfortest"),
+        ("indicator_incident_test_for_test", "incidenttestfortest"),
+        ("indicator_indicator_test_for_test", "indicatortestfortest")
+    ]
+
+    @pytest.mark.parametrize("_id, cliname", VALID_CLINAMES)
+    def test_is_cli_name_and_id_are_matches(self, _id, cliname):
+        with patch.object(StructureValidator, '__init__', lambda a, b: None):
```
Comment (dorschw) [inline demisto_sdk/commands/common/tests/field_validator_test.py:203]: nice

File: demisto_sdk/commands/common/tests/field_validator_test.py [hunk 5]
```diff
@@ -190,6 +190,50 @@ def test_is_cli_name_is_builtin_key_invalid(self, cliname, group):
...
+            structure.old_file = None
+            structure.file_path = "random_path"
+            structure.is_valid = True
+            structure.prev_ver = 'master'
+            structure.branch_name = ''
+            validator = FieldBaseValidator(structure, set(), set())
+            validator.current_file = current_file
+            assert validator.is_cli_name_and_id_are_matches()
+
+    INVALID_CLINAMES = [
```
Comment (dorschw) [inline demisto_sdk/commands/common/tests/field_validator_test.py:216]: please add some cases with upper case strings

**Review by dorschw — COMMENTED** [review]
good job

File: demisto_sdk/commands/common/errors.py [hunk 6]
```diff
@@ -2163,6 +2164,13 @@ def spaces_in_the_end_of_id(item_id: str):
     def spaces_in_the_end_of_name(name: str):
         return f'Content item name "{name}" should not have trailing spaces. Please remove.'
 
+    @staticmethod
+    @error_code_decorator
+    def cli_name_and_id_are_not_matches(cli_name_correct):
+        return "`cliName` and `id` are not matches\n" \
+               "Please fix it so that it is matches except the prefix\n" \
+               f"HINT -> cliName: `{cli_name_correct}`"
```
Comment (dorschw) [inline demisto_sdk/commands/common/errors.py:2172]:
```suggestion
        return 'cliName and id do not match.\n' \
               f'Please change cliName to {cli_name_correct}'
```

File: demisto_sdk/commands/common/hook_validations/field_base_validator.py [hunk 7]
```diff
@@ -214,7 +214,28 @@ def is_valid_cli_name(self) -> bool:
...
+        return self.is_cli_name_is_builtin_key() and self.is_matching_cli_name_regex() \
+            and self.does_cli_name_match_id()
+
+    def does_cli_name_match_id(self):
+        """
+        Checks if the field cliName matched to id.
+        Returns:
+            (bool): True if cliName matched to id, False otherwise.
+        """
+        cli_name = self.current_file.get('cliName', '')
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/field_base_validator.py:226]:
```suggestion
        cli_name = self.current_file['cliName']
```

File: demisto_sdk/commands/common/tests/field_validator_test.py [hunk 8]
```diff
@@ -190,6 +190,50 @@ def test_is_cli_name_is_builtin_key_invalid(self, cliname, group):
...
 
+    VALID_CLINAMES = [
+        ("incident_testfortest", "testfortest"),
+        ("incident_test_for_test", "testfortest"),
+        ("indicator_test_for_test", "testfortest"),
+        ("indicator_incident_test_for_test", "incidenttestfortest"),
+        ("indicator_indicator_test_for_test", "indicatortestfortest")
+    ]
+
+    @pytest.mark.parametrize("_id, cliname", VALID_CLINAMES)
```
Comment (dorschw) [inline demisto_sdk/commands/common/tests/field_validator_test.py:201]:
```suggestion
    @pytest.mark.parametrize('_id, cliname', VALID_CLINAMES)
```

File: demisto_sdk/commands/common/tests/field_validator_test.py [hunk 9]
```diff
@@ -232,9 +276,15 @@ def test_matching_cliname_regex_invalid(self, cliname):
...
-        current_file = {"cliName": cliname, "group": group}
+    VALID_CLINAMES_AND_GROUPS = [
+        ("incident_validind", "validind", GroupFieldTypes.INCIDENT_FIELD),
+        ("validind", "validind", GroupFieldTypes.EVIDENCE_FIELD),
+        ("indicator_validind", "validind", GroupFieldTypes.INDICATOR_FIELD)
+    ]
+
+    @pytest.mark.parametrize("_id, cliname, group", VALID_CLINAMES_AND_GROUPS)
+    def test_is_valid_cliname(self, _id, cliname, group):
+        current_file = {"id": _id, "cliName": cliname, "group": group}
```
Comment (dorschw) [inline demisto_sdk/commands/common/tests/field_validator_test.py:287]:
```suggestion
        current_file = {'id': _id, 'cliName': cliname, 'group': group}
```

**Review by dorschw — APPROVED** [review]

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 10]
```diff
@@ -678,7 +678,7 @@ def test_create_ignored_errors_list(self):
                              "LI101"]
         error_list = validate_manager.create_ignored_errors_list(errors_to_exclude)
         assert ["BA101", "BA102", "BA103", "BA104", "BA105", "BA106", "BA107", "BA108", "BA109",
-                "BA110", 'BA111', "BA112", "BA113", "BA114", "BA115", "BC101", "BC102", "BC103", "BC104"] <= error_list
+                "BA110", 'BA111', "BA112", "BA113", "BA114", "BA115", "BA116", "BC101", "BC102", "BC103", "BC104"] <= error_list
```
Comment (dorschw) [inline demisto_sdk/commands/validate/tests/validators_test.py:681]: odd behavior here, please consult with @DeanArbel whether to fix this `<=` or just add `BA116` like you did here and leave this window broken.

**Comment by idovandijk** [comment]
Are we sure this should always be the case (and has always been in older files too)?

**Pull request merged**